php-frontend/src/main/java/org/sonar/plugins/php/api/tests/PHPCheckVerifier.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.php.api.tests;
+
+import com.google.common.base.Preconditions;
+import com.sonar.sslr.api.typed.ActionParser;
+import com.sonarsource.checks.verifier.SingleFileVerifier;
+import java.io.File;
+import org.sonar.php.parser.PHPParserBuilder;
+import org.sonar.php.tree.visitors.LegacyIssue;
+import org.sonar.plugins.php.api.tree.CompilationUnitTree;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxTrivia;
+import org.sonar.plugins.php.api.visitors.FileIssue;
+import org.sonar.plugins.php.api.visitors.IssueLocation;
+import org.sonar.plugins.php.api.visitors.LineIssue;
+import org.sonar.plugins.php.api.visitors.PHPCheck;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+import org.sonar.plugins.php.api.visitors.PhpFile;
+import org.sonar.plugins.php.api.visitors.PhpIssue;
+import org.sonar.plugins.php.api.visitors.PreciseIssue;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class PHPCheckVerifier {
+
+  private static final ActionParser<Tree> parser = PHPParserBuilder.createParser();
+
+  public static void verify(File sourceFile, PHPCheck check) {
+    verify(new PhpTestFile(sourceFile), check);
+  }
+
+  public static void verifyNoIssue(File sourceFile, PHPCheck check) {
+    verifyNoIssue(new PhpTestFile(sourceFile), check);
+  }
+
+  public static void verify(PhpFile sourceFile, PHPCheck check) {
+    createVerifier(sourceFile, check).assertOneOrMoreIssues();
+  }
+
+  public static void verifyNoIssue(PhpFile sourceFile, PHPCheck check) {
+    createVerifier(sourceFile, check).assertNoIssues();
+  }
+
+  private static SingleFileVerifier createVerifier(PhpFile file, PHPCheck check) {
+    SingleFileVerifier verifier = SingleFileVerifier.create(file.relativePath(), UTF_8);
+    CompilationUnitTree tree = (CompilationUnitTree) parser.parse(file.contents());
+    check.init();
+    for (PhpIssue issue : check.analyze(file, tree)) {
+      Preconditions.checkState(issue.check().equals(check), "Verifier support only one kind of issue " + issue.check() + " != " + check);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Invoke method(s) only conditionally. Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2629)